### PR TITLE
Implement structlog across the major logging components.

### LINKED ISSUE #3309: Implement structlog across the major logging components.

### PR DESCRIPTION
**Pull request by JTCunning**
Fixes #3309.

### CODE REVIEW

File: src/sentry/logging/handlers.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import logging
+
+
+class StructLogHandler(logging.StreamHandler):
+    def __init__(self, *args, **kwargs):
+        from structlog import get_logger
+        from sentry.runner.initializer import configure_structlog
+        super(StructLogHandler, self).__init__(*args, **kwargs)
+        configure_structlog()
```
Comment (mattrobenolt) [inline src/sentry/logging/handlers.py]: Why is `configure_structlog()` being called twice? Here, and inside `initialize_app`.

Comment (JTCunning) [inline src/sentry/logging/handlers.py]: Dartboard cruft, will remove

**Comment by JTCunning** [comment]
Please let me know your opinions on this @getsentry/infrastructure.

File: src/sentry/logging/audit.py [hunk 2]
```diff
@@ -1,74 +0,0 @@
-"""
```
Comment (JTCunning) [inline src/sentry/logging/audit.py]: This is no longer the direction we're heading. Removed the additional logger and it's tests.

**Comment by JTCunning** [comment]
Some real world examples:
```
Traceback (most recent call last):
  File "<console>", line 2, in <module>
ZeroDivisionError: integer division or modulo by zero
name='sentry.shell' level='error' event='intentional'
```
`name='sentry.audit.entry' level='info' event=u'project.remove' actor_label=u'mattrobenolt'`

**Comment by mattrobenolt** [comment]
:cookie:

File: src/sentry/logging/handlers.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+"""
+
+import logging
+
+
+class StructLogHandler(logging.StreamHandler):
+    def __init__(self, *args, **kwargs):
+        from structlog import get_logger
+        super(StructLogHandler, self).__init__(*args, **kwargs)
+        self._structlog = get_logger()
```
Comment (mattrobenolt) [inline src/sentry/logging/handlers.py]: It seems it'd be safe to make this module level then it doesn't have to be an instance on each `StructLogHandler`.

Is there a reason I'm overlooking why this is needed per-instance?

Comment (mattrobenolt) [inline src/sentry/logging/handlers.py]: Also, from my understanding, this relates to the `cache_logger_on_first_use` configuration setting. It's lazy until it's used once.

**Pull request merged**